atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -1451,19 +1492,132 @@ private static SnapshotTransaction unwrapSnapshotTransaction(Transaction transac
         return (SnapshotTransaction) transaction;
     }
 
-    private static class WrappingTestTransactionManagerImpl extends WrappingTestTransactionManager {
-        private final Function<Transaction, Transaction> transactionWrapper;
+    private static class WrappingTestTransactionManagerImpl extends TestTransactionManagerImpl {
+        private final BiFunction<Transaction, AtomicInteger, Transaction> transactionWrapper;
```

Comment:
At some point I'm starting to think that we should just have a special `UnaryOperator<Transaction>` that is a bit more descriptive. To me it's hard to follow what the `AtomicInteger`s are trying to do. Might be worth discussing offline

Reply:
This is removed and we have an interface now for this.